### PR TITLE
Improve edit task and smart apply destination file path computation

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes https://linear.app/sourcegraph/issue/QA-116

## Test plan

It's hard to reproduce actual issue as it depends on the LLM response.
Test scenario which sometimes can reproduce it:

**For "Generate Unit Test":**

1. Opens some workspace, and then file from outside of that workspace
2. Run "Generate Unit Test" for that file
3. Make sure test file is generated properly and the location of the file looks sensible

**For "Smart Apply":**

Case1:

1. Open some workspace
2. Ask Cody to generate some code sample in totally unrelated to the existing workspace (e.g. "quicksort in haskell")
3. Click Smart Apply
4. Make sure Cody created new file for the given snipped and it's name and location looks sensible

Case 2: 

1. Open some workspace, and then unrelated file from outside of that workspace
2. Ask Cody to generate some code related to the opened file
3. Click Smart Apply
4. Make sure Cody appended code to the opened file correctly

Test on both MacOs and Windows.

### CODE REVIEW

File: vscode/src/services/utils/edit-create-file.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    if (!baseDirUri) {
+        return fallbackUri
+    }
+
+    return smartJoinPath(baseDirUri, uri)
+}
+
+export function smartJoinPath(baseDirUri: vscode.Uri, relativeFileUri: string): vscode.Uri {
```
Comment (arafatkatze) [inline vscode/src/services/utils/edit-create-file.ts:43]: No need to export this as you don't call it anywhere?

Comment (pkukielka) [reply vscode/src/services/utils/edit-create-file.ts:43]: It is needed for tests

**Review by arafatkatze — APPROVED** [review]
I tested the PR on mac and it worked fine for all the edge cases of smart apply and edit(that I am aware of). 

Piotr tested it on Windows also(I didn't do that myself  though)
